numpy/core/src/umath/loops.c.src

```diff
@@ -1208,22 +1205,18 @@ NPY_NO_EXPORT void
  * #kind = maximum, minimum#
  * #OP =  >, <#
  **/
-NPY_NO_EXPORT void
+NPY_NO_EXPORT NPY_GCC_OPT_3 void
 @TYPE@_@kind@(char **args, npy_intp *dimensions, npy_intp *steps, void *NPY_UNUSED(func))
 {
-    BINARY_LOOP {
-        const @type@ in1 = *(@type@ *)ip1;
-        const @type@ in2 = *(@type@ *)ip2;
-        if (in1 == NPY_DATETIME_NAT) {
-            *((@type@ *)op1) = in2;
-        }
-        else if (in2 == NPY_DATETIME_NAT) {
-            *((@type@ *)op1) = in1;
-        }
-        else {
-            *((@type@ *)op1) = (in1 @OP@ in2) ? in1 : in2;
-        }
-    }
+    BINARY_LOOP_FAST(@type@, @type@,
+                     if (in1 == NPY_DATETIME_NAT) {
+                         *out = in2;
+                     } else if (in2 == NPY_DATETIME_NAT) {
+                         *out = in1;
+                     } else {
+                         *out = (in1 @OP@ in2) ? in1 : in2;
+                     }
+                     );
```

Comment:
reducing the indent here probably looks nicer

Reply:
Removed as this was one of the datetime loops